res/controllers/Pioneer-DDJ-400-script.js

```diff
@@ -0,0 +1,774 @@
+    }
+    PioneerDDJ400.toggleLight(LightsPioneerDDJ400.beatFx, false);
+});
+
+PioneerDDJ400.beatFxChannel = ignoreRelease(function(_channel, control, _value, _status, group) {
+    "use strict";
+    var enableChannel1 = control === 0x10 || control === 0x14;
+    var enableChannel2 = control === 0x11 || control === 0x14;
+
+    engine.setValue(group, "group_[Channel1]_enable", enableChannel1);
+    engine.setValue(group, "group_[Channel2]_enable", enableChannel2);
+});
+
+PioneerDDJ400.padFxBelowPressed = ignoreRelease(function(channel, control, value, status, group) {
+    "use strict";
+    var groupAbove = group.replace(/\[EffectRack1_EffectUnit(\d+)_Effect(\d+)]/, function(all, unit, effect) {
+        var effectAbove = parseInt(effect) - 4;
+
+        return "[EffectRack1_EffectUnit" + unit + "_Effect" + effectAbove + "]";
+    });
+
+    engine.setValue(groupAbove, "next_effect", value);
+});
+
+PioneerDDJ400.padFxShiftBelowPressed = ignoreRelease(function(channel, control, value, status, group) {
+    "use strict";
+    var groupAbove = group.replace(/\[EffectRack1_EffectUnit(\d+)_Effect(\d+)]/, function(all, unit, effect) {
+        var effectAbove = parseInt(effect) - 4;
+
+        return "[EffectRack1_EffectUnit" + unit + "_Effect" + effectAbove + "]";
+    });
+
+    engine.setValue(groupAbove, "prev_effect", value);
+});
+
+// END BEAT FX
+
+PioneerDDJ400.vuMeterUpdate = function(value, group) {
+    "use strict";
+    var newVal = value * 150;
+
+    switch (group) {
+        case "[Channel1]":
+            midi.sendShortMsg(0xB0, 0x02, newVal);
+            break;
+
+        case "[Channel2]":
+            midi.sendShortMsg(0xB1, 0x02, newVal);
+            break;
+    }
+};
+
+//
+// SAMPLERS
+//
+
+// DJ3730: blink pad when sample playback starts
+PioneerDDJ400.samplerPlayOutputCallbackFunction = function (value, group, control) {
+    if (value === 1) {
+        var curPad = group.match(/^\[Sampler(\d+)\]$/)[1];   // for some reason, using script.samplerRegEx here results in an error under Linux
+        startSamplerBlink((0x97 + (curPad > 8 ? 2 : 0)), (0x30 + ((curPad > 8 ? curPad-8 : curPad)-1)), group);
+    }
+};
+
+PioneerDDJ400.samplerModePadPressed = ignoreRelease(function(_channel, control, _value, status, group) {
+    "use strict";
+    var isLoaded = engine.getValue(group, "track_loaded") === 1;
+
+    if (!isLoaded) {
+        return;
+    }
+    engine.setValue(group, "cue_gotoandplay", 1);
+});
+
+
+PioneerDDJ400.samplerModeShiftPadPressed = function(_channel, _control, value, _status, group){
+    "use strict";
+    if(value == 0) {
+        return; // ignore release
+    }
+    var playing = engine.getValue(group, 'play');
+    // when playing stop and return to start/cue point
+    if(playing > 0){
+        engine.setValue(group, 'cue_gotoandstop', 1);
+    }
+    else{ // load selected track
+        // engine.setValue(group, 'LoadSelectedTrack', 1);
+    }
+};
+
+const TimersPioneerDDJ400 = {};
```

Comment:
`const` is also not supported, use `var`

Reply:
Done